commit 3d62ad9a0158a1fe56cf3cff9fc49bfb6d504b27
Author: florian <florian@a5019735-40e9-0310-863c-91ae7b9d1cf9>
Date:   Mon Nov 24 22:30:49 2014 +0000

    Change the testcase to not depend on the type of 'this' in the output.
    GCC 4.9.2 and GCC 4.8.x have different opinions about that variable's type.
    
    
    git-svn-id: svn://svn.valgrind.org/valgrind/trunk@14780 a5019735-40e9-0310-863c-91ae7b9d1cf9

diff --git a/memcheck/tests/demangle.cpp b/memcheck/tests/demangle.cpp
index 6fb2b98..ff15fe1 100644
--- a/memcheck/tests/demangle.cpp
+++ b/memcheck/tests/demangle.cpp
@@ -16,7 +16,7 @@ class magic {
 public:
   T xyzzy(T *p)
   {
-    return (new abc::def<int,typeof(this)>)->xyzzy(p, 0);
+    return (new abc::def<int,typeof(*this)>)->xyzzy(p, 0);
   }
 };
 
diff --git a/memcheck/tests/demangle.stderr.exp b/memcheck/tests/demangle.stderr.exp
index b79c638..2124517 100644
--- a/memcheck/tests/demangle.stderr.exp
+++ b/memcheck/tests/demangle.stderr.exp
@@ -1,6 +1,6 @@
 
 Conditional jump or move depends on uninitialised value(s)
-   at 0x........: abc::def<int, magic<int>* const>::xyzzy(int*, magic<int>* const*) (demangle.cpp:9)
+   at 0x........: abc::def<int, magic<int> >::xyzzy(int*, magic<int>*) (demangle.cpp:9)
    by 0x........: magic<int>::xyzzy(int*) (demangle.cpp:19)
    by 0x........: main (demangle.cpp:27)
 
commit 79a59b53abd68d3d5df2f77bfbe23d6c3eab0e4b
Author: mjw <mjw@a5019735-40e9-0310-863c-91ae7b9d1cf9>
Date:   Wed Jun 3 09:52:00 2015 +0000

    Run memcheck/tests/demangle with -q.
    
    The interesting part is the demangled backtrace in the error message.
    Suppress the memory allocation/blocks summary which can differ slightly
    depending on the underlying arch/libs.
    
    git-svn-id: svn://svn.valgrind.org/valgrind/trunk@15308 a5019735-40e9-0310-863c-91ae7b9d1cf9

diff --git a/memcheck/tests/demangle.stderr.exp b/memcheck/tests/demangle.stderr.exp
index 2124517..6e7467d 100644
--- a/memcheck/tests/demangle.stderr.exp
+++ b/memcheck/tests/demangle.stderr.exp
@@ -1,16 +1,5 @@
-
 Conditional jump or move depends on uninitialised value(s)
    at 0x........: abc::def<int, magic<int> >::xyzzy(int*, magic<int>*) (demangle.cpp:9)
    by 0x........: magic<int>::xyzzy(int*) (demangle.cpp:19)
    by 0x........: main (demangle.cpp:27)
 
-
-HEAP SUMMARY:
-    in use at exit: 6 bytes in 3 blocks
-  total heap usage: 3 allocs, 0 frees, 6 bytes allocated
-
-For a detailed leak analysis, rerun with: --leak-check=full
-
-For counts of detected and suppressed errors, rerun with: -v
-Use --track-origins=yes to see where uninitialised values come from
-ERROR SUMMARY: 1 errors from 1 contexts (suppressed: 0 from 0)
diff --git a/memcheck/tests/demangle.vgtest b/memcheck/tests/demangle.vgtest
index f6ae038..b65b6d3 100644
--- a/memcheck/tests/demangle.vgtest
+++ b/memcheck/tests/demangle.vgtest
@@ -1 +1,2 @@
 prog: demangle
+vgopts: -q
